ast/path.go

```diff
@@ -0,0 +1,42 @@
+package ast
+
+import (
+	"bytes"
+	"fmt"
+)
+
+type Path []PathElement
+
+type PathElement interface {
+	isPathElement()
+}
+
+var _ PathElement = PathIndex(0)
+var _ PathElement = PathName("")
+
+func (path Path) String() string {
+	var str bytes.Buffer
+	for i, v := range path {
+
+		switch v := v.(type) {
+		case PathIndex:
+			str.WriteString(fmt.Sprintf("[%d]", v))
+		case PathName:
+			if i != 0 {
```

Comment:
bug if path starts with number?

eg `.0.foo.bar`?

Reply:
expected is `[0].foo.bar` ? 🤔 
I think number is not coming at head of path.
so, this PR can't validate element value.
1. don't place number at head
2. don't contains `nil` value